Style-Guide/Code-Layout-and-Formatting.md

```diff
@@ -68,35 +67,57 @@ A special case is made for two-letter acronyms in which both letters are capital
 
+## Proper Structure of Functions
 
-#### Always Start With CmdletBinding
+### Always Start With CmdletBinding
 
 All of your scripts or functions should start life as something like this snippet:
 
+```powershell
+[CmdletBinding()]
+param()
+process {}
+end {}
 ```
-[CmdletBinding()]param()
-process{}
-end{}
+
+You can always delete or ignore one of the blocks (or add the `begin` block), add parameters and necessary valiation and so on, but you should **avoid** writing scripts or functions without `[CmdletBinding()]`, and you should always at least _consider_ making it take pipeline input.
+
+## Open braces on the same line
+
+This can be considered a matter of consistency; several common cmdlets in PowerShell take script blocks as _parameters_ (e.g., `ForEach-Object`), and in these cases it is functionally impossible to place the opening brace on a new line _without_ use of a line-continuator (i.e., ``` ` ```, a backtick), which should generally be avoided.
+
+```powershell
+$Data | ForEach-Object {
+    $_.Item -as [int]
+}
 ```
 
-You can always delete or ignore one of the blocks (or add the `begin` block), add parameters and so on, but you should avoid writing scripts or functions without CmdletBinding, and you should always at least _consider_ making it take pipeline input.
+v.s.
+
+```powershell
+foreach ($Entry in $Data)
+{
+    $Entry.Item -as [int]
+}
+```
+
+As such, both native keywords and function parameters should include opening braces on the _same_ line.
+
+Code folding is also nicer in many editors.
 
-#### Open braces on the same line
-Code folding is nicer in many editors.
-(TODO: This is in discussion in [#24](https://github.com/PoshCode/PowerShellPracticeAndStyle/issues/24))
+## Closing braces always on their own line
 
-#### Closing braces always on their own line
-Because that's how they're supposed to be!
-(TODO: This is in discussion in [#24](https://github.com/PoshCode/PowerShellPracticeAndStyle/issues/24))
+Once again, this makes code-folding much more sensible in many editors.
 
-#### Prefer: param() begin, process, end
-That's the order PowerShell will execute it in
-(TODO)
+The exception to this rule may be in cases where the script block is a parameter, and further parameters must still be added. However, in the interests of improving code-folding, readability, and maintainability, placing such parameters _before_ the script block parameter should be considered, where possible.
 
+### Prefer: param() begin, process, end
 
-#### Indentation
+Having a script written in the order of execution makes its intent more clear. There is no functional purpose to having `begin` be declared _after_ `process`, and although it will still be executed in the correct order, it should be considered that writing in such a fashion significantly detracts from the overall comprehensibility of a script.
```

Comment:
Very long sentence, please break up. 

Remove "it should be considered that" (it's redundant). Try for a simpler wording of "writing in such a fashion" and "significantly detracts for the overall comprehensibility"